compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1092,6 +1092,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
             // this as a special case.
             return self.lower_fn_body(decl, |this| {
                 if attrs.iter().any(|a| a.name_or_empty() == sym::rustc_intrinsic) {
+                    let span = this.lower_span(span);
```

Comment:
Unlikely to have any effect anywhere, as this is only for intrinsics, but to pacify the new debug assert